bread/utils/links.py

```diff
@@ -58,30 +57,52 @@ class ModelHref(LazyHref):
             models.Person, "edit", kwargs={"pk": hg.C("object.pk")}
         ).resolve(context)
 
+    return_to_current: will add a URL query parameter "next=<current_url" to the generated URL
+
     """
 
-    def __init__(self, model, name, *args, **kwargs):
+    def __init__(
+        self,
+        model: Union[models.Model, hg.Lazy],
+        name: str,
+        *args,
+        return_to_current: bool = False,
+        **kwargs
+    ):
+
         # if this is an instance of a model, we can extract the pk URL argument directly
         # TODO: instance-specific routes which don't use the pk argument will fail
-        if isinstance(model, models.Model) and "pk" not in kwargs.get("kwargs", {}):
-            if "kwargs" not in kwargs:
-                kwargs["kwargs"] = {}
-            kwargs["kwargs"]["pk"] = model.pk
-
         if isinstance(model, hg.Lazy):
             url = hg.F(lambda c: model_urlname(hg.resolve_lazy(model, c), name))
         else:
             url = model_urlname(model, name)
+        if return_to_current:
+            if "query" not in kwargs:
+                kwargs["query"] = {}
+            kwargs["query"]["next"] = hg.C("request.get_full_path")
 
         super().__init__(url, *args, **kwargs)
 
+    @staticmethod
+    def from_object(object: Union[models.Model, hg.Lazy], name: str, *args, **kwargs):
+        """
+        name: string which denotes an object action name as generated
+              by bread.utils.urls.model_urlname
+        object: instance of a Model, can be lazy
+        """
+
+        if "kwargs" not in kwargs:
+            kwargs["kwargs"] = {}
+        kwargs["kwargs"]["pk"] = object.pk
```

Comment:
Hmm `kwargs["kwargs"]`... Could we maybe rename one of these? Something like `kwargs["url_path_parameters"]`? (This would be a larger refactoring, I know. But the "kwargs" that are used for url path parameters was quite confusing for me when I first saw it.

Reply:
I would not really recommend it, although I know it is a bit confusing. The root of the problem is that Django has a function parameter called ```kwargs``` at https://docs.djangoproject.com/en/dev/ref/urlresolvers/#reverse. I would like to keep the name-compatatiliby with the original function, especially inside ```bread.utils.urls.reverse```. And the idea of the ```links``` module would be to abstract some of the details aways, at least for ```ModelHref```.